src/d3a/models/strategy/__init__.py

```diff
@@ -330,6 +330,33 @@ def post_offer(self, market, replace_existing=True, **offer_kwargs) -> Offer:
 
         return offer
 
+    def post_first_offer(self, market, energy_kWh, initial_energy_rate):
+        if not all(offer.seller != self.owner.name for offer in market.get_offers().values()):
```

Comment:
```suggestion
        if any(offer.seller != self.owner.name for offer in market.get_offers().values()):
```
Wouldn't it be simpler in order to avoid reverse of logic check?

Reply:
Good idea, will apply it to both post_first_bid and post_first_offer, thanks!